src/BlazorWebFormsComponents/BaseWebFormsComponent.cs

```diff
@@ -226,21 +226,27 @@ public ValueTask DisposeAsync()
 
 		public BaseWebFormsComponent ()
 		{
+			// Get Access to the ComponentBase fields we need to wrap every component in a CascadingValue
+			var _renderFragmentField = typeof(ComponentBase).GetField("_renderFragment", BindingFlags.NonPublic | BindingFlags.Instance);
+			var _hasPendingQueuedRenderField = typeof(ComponentBase).GetField("_hasPendingQueuedRender", BindingFlags.NonPublic | BindingFlags.Instance);
+			var _hasNeverRenderedField = typeof(ComponentBase).GetField("_hasNeverRendered", BindingFlags.NonPublic | BindingFlags.Instance);
+
+			// Create a RenderFragment to go into the CascadingValue
 			RenderFragment _standardRender = builder2 => {
-				_hasPendingQueuedRender = false;
-				_hasNeverRendered = false;
+				_hasPendingQueuedRenderField.SetValue(this,false);

```

Comment:
Curious, why is setting `_hasPendingQueuedRenderField` and `_hasNeverRenderedField` necessary?

Reply:
It's part of the standard renderFragment from ComponentBase - I don't see any reason to change the standard.

Reply:
https://github.com/dotnet/aspnetcore/blob/1e735afa929e958dd62a7e68db990bbacd675a25/src/Components/Components/src/ComponentBase.cs#L40

Reply:
I see. Why do you need to set them in your renderFragment action if they are set by the `ComponentBase._renderFragment`? When the rendering has been started, which it has at this point, then it should not matter?

Reply:
Ahh nevermind, you are not reusing the _renderFragment, you are replacing it. Yep, it looks right :)

Reply:
OK, but maybe you can minimize required reflection of the base code, and just grab the renderFragment and call that, e.g.:

```csharp
var _renderFragmentField = typeof(ComponentBase).GetField("_renderFragment", BindingFlags.NonPublic | BindingFlags.Instance);

var baseRenderFragment = _renderFragmentField.GetValue(this);

// Override the default RenderFragment with our Special Sauce version
_renderFragmentField.SetValue(this, (RenderFragment)(builder =>
{
	builder.OpenComponent(1, typeof(CascadingValue<BaseWebFormsComponent>));
	builder.AddAttribute(2, nameof(CascadingValue<BaseWebFormsComponent>.Name), nameof(ParentComponent));
	builder.AddAttribute(3, nameof(CascadingValue<BaseWebFormsComponent>.Value), this);
	builder.AddAttribute(4, nameof(CascadingValue<BaseWebFormsComponent>.Fixed), true);
	builder.AddAttribute(5, nameof(CascadingValue<BaseWebFormsComponent>.ChildContent), baseRenderFragment);
	builder.CloseComponent();
}));
```

Reply:
Yes, I'll try that out and see - it would certainly be better to just use the default rather than recreating it.

Reply:
ps. i like how reflection allows us to completely ignore the fact that the field _renderFragment is readonly :)

Reply:
I'm not 100% sure it does - we are, after all, setting it in the constructor, which is allowed.

I haven't tried setting it elsewhere, but I would hope it would fail.

Reply:
Ahh yeah, lets hope thats the case.